lib/src/lock_file.dart

```diff
@@ -98,6 +107,7 @@ class LockFile {
     Uri sourceUrl;
     if (filePath != null) sourceUrl = p.toUri(filePath);
     var parsed = loadYamlNode(contents, sourceUrl: sourceUrl);
+    final windowsLineEndings = contents.contains('\r\n');
```

Comment:
If we wanted to be a bit more robust we do `count(contents, '\r\n') > count(contents, '\n')`, using:
```dart
int count(String s, Pattern p) {
  int c = 0;
  int i = -1;
  while ((i = s.indexOf(p, i + 1)) != -1) {
    c++;
  }
  return c;
}
```

So if there is more CRLF's than LFs we use CRLF, otherwise we use LF.

It's hard to imagine things getting mixed.. But maybe if someone edits the `pubspec.lock` or manually resolves a merge conflict? (I suspect most editors would normalize line endings, but just in case it might be nice).

Reply:
We at least have to make it `count(contents, '\r\n') > count(contents, '[^\r]\n')`.

But yes, it would probably be safer.

Reply:
Done